spec/services/bookings/gitis/contact_spec.rb

```diff
@@ -115,4 +114,67 @@
       it { is_expected.to be true }
     end
   end
+
+  describe 'writing' do
+    describe "#attributes_for_create" do
+      let(:contact) do
+        described_class.new.tap do |c|
+          c.contactid     = SecureRandom.uuid
+          c.first_name    = "Test"
+          c.last_name     = "User"
+          c.email         = 'testing@testaddress.education.gov.uk'
+          c.date_of_birth = Date.parse('1980-01-01')
+          c.phone         = '01234 567890'
+          c.building      = 'My Building'
+          c.street        = 'Test Street'
+          c.town_or_city  = 'Test Town'
+          c.county        = 'Test County'
+          c.postcode      = 'MA1 1AM'
+        end
+      end
+
+      subject { contact.attributes_for_create }
+      it { is_expected.not_to include('contactid') }
+      it { is_expected.to include('firstname') }
+      it { is_expected.to include('lastname') }
+      it { is_expected.to include('emailaddress1') }
+      it { is_expected.to include('emailaddress2') }
+      it { is_expected.to include('telephone2') }
+      it { is_expected.to include('statecode') }
+      it { is_expected.to include('dfe_channelcreation') }
+    end
+
+    describe "#attributes_for_update" do
+      let(:attrs) do
+        attributes_for :gitis_contact, :persisted, dfe_channelcreation: nil
+      end
+
+      let(:contact) { Bookings::Gitis::Contact.new attrs }
+      subject { contact.attributes_for_update }
+
+      context 'when unmodified' do
+        it { is_expected.not_to include('contactid') }
+        it { is_expected.not_to include('statecode') }
+        it { is_expected.not_to include('dfe_channelcreation') }
+        it { is_expected.to include('emailaddress2') }
```

Comment:
adding to this context block:
```
        it { is_expected.to include('firstname') }
        it { is_expected.to include('lastname') }
        it { is_expected.to include('telephone2') }
```
still passes.
Might be something to do with `contact.persisted? #=> false` even though we're building it with the `:persisted` trait?